### ISSUE TITLE
IDE0040 (Add accessibility modifiers) is not enforced for types nested in interfaces

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#74245]
**Version Used**: 

```
Microsoft Visual Studio Professional 2022
Version 17.10.0 Preview 6.0
VisualStudio.17.Preview/17.10.0-pre.6.0+34825.169
Microsoft .NET Framework
Version 4.8.09037
```

**Steps to Reproduce**:

Set the option in editorconfig:
```
dotnet_style_require_accessibility_modifiers = for_non_interface_members:error
```

Compile the following code:
```cs
public class MyClass
{
  public class Nested
  {
    void Method() { } // IDE0040 (correct)
  }
}

public interface IMyInterface
{
  public class Nested
  {
    void Method() { } // missing IDE0040
  }
}
```

**Diagnostic Id**:

```IDE0040 Add accessibility modifiers```

**Expected Behavior**:
`IDE0040` is reported for both `Nested.Method` methods

**Actual Behavior**:
`IDE0040` is only reported for a type nested in another class, but not in an interface

**Notes**:
`for_non_interface_members` was added in order not to enforce default `public` modifiers for interface members and keep them consistent with the only allowed declaration style in older C# versions, see https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0040?view=vs-2022 and https://github.com/dotnet/roslyn/blob/f61885262640ef325336468ae2368e26217471da/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/AccessibilityModifiersRequired.cs#L15-L19

Since nested types' members are neither `public`, nor "interface members", nor were possible to declare in older C# versions, `for_non_interface_members` option should include them

**Comment by CyrusNajmabadi** [comment]
We would take a targeted community pr here.

**Comment by justurbo** [comment]
Well, this _minor_ change really broke our build

**Comment by CyrusNajmabadi** [comment]
@justurbo you can disable the rule, suppress the diagnostic, or update the code to follow the requirements of the rule. thanks.
